lib/senders/email.js

```diff
@@ -226,10 +226,18 @@ module.exports = function (log) {
 
     var localized = this.localize(message)
 
-    if (message.flowBeginTime && message.flowId) {
-      message.headers['X-Flow-Id'] = message.flowId
-      message.headers['X-Flow-Begin-Time'] = message.flowBeginTime
-    }
+    const headers = Object.assign(
+      {
+        'Content-Language': localized.language,
+        'X-Template-Name': message.template
+      },
+      message.headers,
+      optionalHeader('X-Device-Id', message.deviceId),
+      optionalHeader('X-Flow-Id', message.flowId),
+      optionalHeader('X-Flow-Begin-Time', message.flowBeginTime),
+      optionalHeader('X-Service-Id', message.service),
+      optionalHeader('X-Uid', message.uid)
```

Comment:
Previously these were each being set for some subset of the templates. To maximise the data we send to amplitude, I now set them aggressively whenever we have values for them.